### PR TITLE
Add size limits to RandomScale transform

### PR DESCRIPTION
**Pull request by ShahJahanIshaq**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

During training, the RandomScale transform randomly scaled the images in the datasets; however, in some datasets, there are mixtures of small and big images, which, when scaled randomly, may lead to extremely small or extremely large images.

Therefore, this PR implements the logic of taking the minimum and maximum side lengths (`side_limits`) as arguments and then using them to calculate the scaling factor, so the randomly scaled images always stay within the specified reasonable limits.

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/data/transforms/general_transforms.py [hunk 1]
```diff
@@ -175,7 +183,14 @@ def __call__(self, data: dict) -> dict:
             (polys)
         """
         if random.random() < self._p:
-            scale = np.random.uniform(*self._range)
+            size = data["image"].shape[:2]
```
Comment (hadipash) [inline mindocr/data/transforms/general_transforms.py:186]: Move `size` inside `if` statement as it is only used there.

File: mindocr/data/transforms/general_transforms.py [hunk 2]
```diff
@@ -157,11 +157,19 @@ class RandomScale:
...
-    def __init__(self, scale_range: Union[tuple, list], p: float = 0.5, **kwargs):
+    def __init__(
+        self,
+        scale_range: Union[tuple, list],
+        size_limits: Union[tuple, list] = None,
+        p: float = 0.5,
+        **kwargs,
+    ):
         self._range = scale_range
+        self._size_limits = size_limits
```
Comment (hadipash) [inline mindocr/data/transforms/general_transforms.py:172]: Sort both `self._range` and `self._size_limits` to ensure the correct order (min, max).

**Comment by hadipash** [comment]
@colawyee FCENet's original scaling transformation required size limits. This PR adds this functionality so that you can integrate limits in FCENet if you find it necessary.

File: mindocr/data/transforms/general_transforms.py [hunk 3]
```diff
@@ -157,11 +157,19 @@ class RandomScale:
...
+        size_limits: (min_side_len, max_side_len) size limits. Default: [].
         p: probability of the augmentation being applied to an image.
     """
 
-    def __init__(self, scale_range: Union[tuple, list], p: float = 0.5, **kwargs):
-        self._range = scale_range
+    def __init__(
+        self,
+        scale_range: Union[tuple, list],
+        size_limits: Union[tuple, list] = [],
```
Comment (hadipash) [inline mindocr/data/transforms/general_transforms.py:167]: be careful of [mutable default arguments](https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/)

**Review by hadipash — APPROVED** [review]

**Review by Songyuanwei — APPROVED** [review]

**Pull request merged**